### PR TITLE
feat: dokku_http_auth library

### LINKED ISSUE #133: How should I configure HTTP Basic Auth?
I have to protect some internal tools like `loki` and would like to use the https://github.com/dokku/dokku-http-auth plugin. 
I can easily install it via the `dokku_plugins` module / library, but how am I supposed to enable it per app using Ansible? Running the commands directly using the `command` module seems like a workaround and I'd like to know what's the recommended way to achieve this.

### PR DESCRIPTION
**Pull request by fr3fou**
Closes #133

### CODE REVIEW

File: library/dokku_http_auth.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    report = response[0].split(":")[1]
+    return report.strip() == "true", error
+
+
+def dokku_http_auth_present(data):
+    is_error = True
+    has_changed = False
+    meta = {"present": False}
+
+    enabled, error = dokku_http_auth_enabled(data)
```
Comment (ltalirz) [inline library/dokku_http_auth.py:72]: What if an error is returned here?
It seems to me this is currently silently ignored...

File: library/dokku_http_auth.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    return (is_error, has_changed, meta)
+
+
+def dokku_http_auth_absent(data=None):
+    is_error = True
+    has_changed = False
+    meta = {"present": True}
+
+    enabled, error = dokku_http_auth_enabled(data)
```
Comment (ltalirz) [inline library/dokku_http_auth.py:97]: same here

**Review by ltalirz — APPROVED** [review]
Thanks a lot for this contribution @fr3fou !

Just one little nitpick; then this is good to go.

**Review by ltalirz — CHANGES_REQUESTED** [review]
sorry it's late - intended to "request changes"...

File: library/dokku_http_auth.py (hunk 1 above)
Comment (fr3fou) [reply library/dokku_http_auth.py:72]: oops, must have forgotten this

**Pull request merged**